salt/modules/ssh.py

```diff
@@ -149,15 +149,8 @@ def _replace_auth_key(
                     # Commented Line
                     lines.append(line)
                     continue
-                comps = line.split()
-                if len(comps) < 2:
-                    # Not a valid line
-                    lines.append(line)
-                    continue
-                key_ind = 1
-                if comps[0][:4:] not in ['ssh-', 'ecds']:
-                    key_ind = 2
-                if comps[key_ind] == key:
+                comps = re.findall(r'((.*)\s)?(ssh-[a-z0-9-]+|ecdsa-[a-z0-9-]+)\s([a-zA-Z0-9+/]+={0,2})(\s(.*))?', line)
```

Comment:
I really tried to follow the idea of the original code. Which just split the line by spaces and tried to match 1st part for (ssh-|ecds). If it matched, then treat the line as if second part is the key, if it did not match, use third part as the key. 

`((.*)\s)?` the line may start with options. it may contain pretty much anything, including quotes, escaped quotes, and so on, followed by a space.

`(ssh-[a-z0-9-]+|ecdsa-[a-z0-9-]+)\s` next is the key type, which really the only known part of the line. Presently it has to start with ssh- or ecds-. Followed by a space.

`([a-zA-Z0-9+/]+={0,2})` now we have to match the key. It is base64 encoded.

`(\s(.*))?` the last is a comment. Which again is pretty much free-form. 

So in short, now I'm looking for `(ssh-|ecdsa-)` as i know that is followed by a space and a base64 encoded key.

To compare, this is how openssh itself does it: https://github.com/openssh/openssh-portable/blob/52763dd3fe0a4678dafdf7aeb32286e514130afc/auth2-pubkey.c#L774 ... I really don't feel like porting that to python and not causing any damage myself.